src/Markdig/Parsers/BlockProcessor.cs

```diff
@@ -16,165 +17,159 @@ namespace Markdig.Parsers
     /// </summary>
     public class BlockProcessor
     {
-        private readonly BlockProcessor? root;
         private int currentStackIndex;
-        private readonly BlockParserStateCache parserStateCache;
-        private int originalLineStart = 0;
-
-        private BlockProcessor(BlockProcessor root, bool trackTrivia = false)
-        {
-            // These properties are not changing between a parent and a children BlockProcessor
-            this.root = root;
-            this.parserStateCache = root.parserStateCache;
-            TrackTrivia = trackTrivia;
-            Document = root.Document;
-            Parsers = root.Parsers;
-
-            // These properties are local to a state
-            OpenedBlocks = new List<Block>();
-            NewBlocks = new Stack<Block>();
-        }
+        private int originalLineStart;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="BlockProcessor" /> class.
+        /// True to parse trivia such as whitespace, extra heading characters and unescaped
+        /// string values.
         /// </summary>
-        /// <param name="document">The document to build blocks into.</param>
-        /// <param name="parsers">The list of parsers.</param>
-        /// <param name="context">A parser context used for the parsing.</param>
-        /// <param name="trackTrivia">Whether to parse trivia such as whitespace, extra heading characters and unescaped string values.</param>
-        /// <exception cref="ArgumentNullException">
-        /// </exception>
-        public BlockProcessor(MarkdownDocument document, BlockParserList parsers, MarkdownParserContext? context, bool trackTrivia = false)
-        {
-            if (document is null) ThrowHelper.ArgumentNullException(nameof(document));
-            if (parsers is null) ThrowHelper.ArgumentNullException(nameof(parsers));
-            TrackTrivia = trackTrivia;
-            parserStateCache = new BlockParserStateCache(this);
-            Document = document;
-            document.IsOpen = true;
-            Parsers = parsers;
-            Context = context;
-            OpenedBlocks = new List<Block>();
-            NewBlocks = new Stack<Block>();
-            root = this;
-            Open(document);
-        }
+        public bool TrackTrivia { get; private set; }
 
         public bool SkipFirstUnwindSpace { get; set; }
 
+        private bool ContinueProcessingLine { get; set; }
+
         /// <summary>
-        /// Gets the new blocks to push. A <see cref="BlockParser"/> is required to push new blocks that it creates to this property.
```

Comment:
Why are the order of the properties are all changed in this file? Could this be done in a separate PR?

Reply:
Oh yeah I can try to revert this part. I was doing it to triple-check all state was getting reset

Reply:
Changed back to original order